pkg/operator/ceph/object/admin_test.go

```diff
@@ -126,37 +127,132 @@ this line can't be parsed as json
 	}
 
-	t.Run("no network provider - we run the radosgw-admin command from the operator", func(t *testing.T) {
+	t.Run("multus - remote execution", func(t *testing.T) {
+		objContext.clusterInfo.NetworkSpec = v1.NetworkSpec{Provider: "multus"}
+
+		remoteEx := new(execmock.RemotePodCommandExecutor)
+		remoteEx.On(
+			"ExecCommandInContainerWithFullOutputWithTimeout", mock.Anything, "rook-ceph-mgr", "cmd-proxy", "mycluster",
+			"radosgw-admin", "zone", "get",
+		).Return(
+			"GOOD", "", nil,
+		)
+
+		objContext.Context.RemoteExecutor = remoteEx
+
+		out, err := RunAdminCommandNoMultisite(objContext, false, []string{"zone", "get"}...)
+		assert.NoError(t, err)
+		assert.Equal(t, "GOOD", out)
+	})
+
+	t.Run("no multus - local execution", func(t *testing.T) {
+		objContext.clusterInfo.NetworkSpec = v1.NetworkSpec{Provider: ""}
+
+		executor := &exectest.MockExecutor{
+			MockExecuteCommandWithTimeout: func(timeout time.Duration, command string, args ...string) (string, error) {
+				if args[0] == "zone" {
+					return "GOOD", nil
+				}
+				return "", nil
+			},
+		}
+
+		objContext.Context.Executor = executor
+
+		out, err := RunAdminCommandNoMultisite(objContext, false, []string{"zone", "get"}...)
+		assert.NoError(t, err)
+		assert.Equal(t, "GOOD", out)
+	})
+
+	t.Run("no multus - retry with remote execution when local execution fails", func(t *testing.T) {
+		objContext.clusterInfo.NetworkSpec = v1.NetworkSpec{Provider: ""}
+
 		executor := &exectest.MockExecutor{
 			MockExecuteCommandWithTimeout: func(timeout time.Duration, command string, args ...string) (string, error) {
 				if args[0] == "zone" {
-					return `{
-		"id": "237e6250-5f7d-4b85-9359-8cb2b1848507",
-		"name": "realm-a",
-		"current_period": "df665ecb-1762-47a9-9c66-f938d251c02a",
-		"epoch": 2
-	}`, nil
+					return "", fmt.Errorf("induced error")
 				}
 				return "", nil
 			},
 		}
 
+		remoteEx := new(execmock.RemotePodCommandExecutor)
+		remoteEx.On(
+			"ExecCommandInContainerWithFullOutputWithTimeout", mock.Anything, "rook-ceph-mgr", "cmd-proxy", "mycluster",
+			"radosgw-admin", "zone", "get",
+		).Return(
+			"GOOD", "", nil,
+		)
+
 		objContext.Context.Executor = executor
-		_, err := RunAdminCommandNoMultisite(objContext, true, []string{"zone", "get"}...)
+		objContext.Context.RemoteExecutor = remoteEx
+
+		out, err := RunAdminCommandNoMultisite(objContext, false, []string{"zone", "get"}...)
 		assert.NoError(t, err)
+		assert.Equal(t, "GOOD", out)
 	})
 
-	t.Run("with multus - we use the remote executor", func(t *testing.T) {
+	t.Run("multus - JSON parsing", func(t *testing.T) {
 		objContext.clusterInfo.NetworkSpec = v1.NetworkSpec{Provider: "multus"}
-		_, err := RunAdminCommandNoMultisite(objContext, true, []string{"zone", "get"}...)
-		assert.Error(t, err)
 
-		// This is not the best but it shows we go through the right codepath
-		assert.EqualError(t, err, "no pods found with selector \"rook-ceph-mgr\"")
```

Comment:
If we are relying on this methodology more, it seemed good to do better testing for it. That meant converting the remote executor struct to an interface that we could use dependency injection for mocking.